### PR TITLE
vlq

### PR DESCRIPTION
**Pull request by magcius**

### CODE REVIEW

**Comment by wgreenberg** [comment]
btw do you think this is still worth implementing, along with the other space-saving index changes? imo the index file's pretty slim as-is, but these do seem like easy wins

**Comment by magcius** [comment]
The sheepfile index is something like 5MB total, which is well below what I would care about (the JS package for noclip clocks in at around 10MB currently, I believe, not even including the WASM modules). So, not urgently needed to spin a new sheepfile version or anything. VLQ also isn't a guaranteed win, it chunks integers into 7-bit bytes, so anything above 28 bits would require 5 bytes to represent. But for all other cases, it's break-even or lower, making it a good candidate for fields like file lengths that average on the smaller side.

tl;dr I'm mildly curious how well VLQ would work on our index just for science's sake, but I don't care about deploying it.
